From 8fc8e4b45996a50565cc883362e403116760d329 Mon Sep 17 00:00:00 2001
From: jacob-delgado <jacob.delgado@volunteers.acasi.info>
Date: Thu, 17 Mar 2022 11:28:52 -0600
Subject: Verify distroless base images before building (#37960)

Based on
https://www.cncf.io/blog/2021/09/14/how-to-secure-containers-with-cosign-and-distroless-images/
we should verify the distroless images before we build images from them.
---
 tools/istio-docker.mk | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/istio-docker.mk b/tools/istio-docker.mk
index 3b14336487..2dbe10474d 100644
--- a/tools/istio-docker.mk
+++ b/tools/istio-docker.mk
@@ -250,6 +250,7 @@ build.docker.install-cni: cni/deployments/kubernetes/Dockerfile.install-cni
 build.docker.base: docker/Dockerfile.base
 	$(DOCKER_RULE)
 build.docker.distroless: docker/Dockerfile.distroless
+	grep gcr.io/distroless ./docker/Dockerfile.distroless  | awk '{print $$2}' | xargs -L1 cosign verify --key https://raw.githubusercontent.com/GoogleContainerTools/distroless/main/cosign.pub
 	$(DOCKER_RULE)
 
 # VM Base images
-- 
2.35.3

